clustermesh: add hostname endpointslice sync (#31814)
cilium/cilium: Documentation/network/clustermesh/services.rst

To have Cilium discover remote clusters endpoints of a Global Service
from DNS or any third party controllers, enable synchronization by adding
the annotation ``service.cilium.io/global-sync-endpoint-slices: "true"``.

> This will allow Cilium to create k8s EndpointSlices that belong a remote cluster

Reviewer: ```suggestion
This will allow Cilium to create Kubernetes EndpointSlices belonging to a remote cluster
```

-----------------

- This is a beta feature, you may experience bugs or shortcomings.
- Hostnames are synchronized as is without any form of conflict resolution

>   mechanisms. This means that multiple StatefulSets with a a single governing

Reviewer: ```suggestion
  mechanisms. This means that multiple StatefulSets with a single governing
```

  Service that synchronize EndpointSlices across multiple clusters should have

>   different names. For instance, this can be overcome by adding the cluster name
>   in the StatefulSet name (i.e.: ``cluster1-my-statefulset`` instead of

Reviewer: ```suggestion
  different names. For instance, you can add the cluster name
  to the StatefulSet name (``cluster1-my-statefulset`` instead of
```
